packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkBlockchain.ts

```diff
@@ -277,10 +298,17 @@ export class ForkBlockchain implements PBlockchain {
     ) {
       return undefined;
     }
-    const transaction = new Transaction(rpcToTxData(rpcTransaction), {
+
+    const chainId = this._common.chainId();
+    const txData = rpcToTxData(rpcTransaction);
+    const senderPubKey = new Transaction(txData).getSenderPublicKey();
+
+    const transaction = new ForkTransaction(chainId, senderPubKey, txData, {
```

Comment:
This `chainId` is not the right one. Here we need the remote network's chain id, but we are using the local one here.